src/org/opendatakit/briefcase/transfer/TransferForms.java

```diff
@@ -15,56 +15,74 @@
  */
 package org.opendatakit.briefcase.transfer;
 
+import static java.util.function.Function.identity;
 import static java.util.stream.Collectors.toList;
 import static java.util.stream.Collectors.toMap;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-import java.util.function.Consumer;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.opendatakit.briefcase.model.BriefcasePreferences;
 import org.opendatakit.briefcase.model.FormStatus;
+import org.opendatakit.briefcase.reused.Pair;
 
 /**
  * This class represents a set of forms to be pulled/pushed. It manages the
  * selection of those forms, as well as merging changes in the forms cache.
  */
-public class TransferForms {
+public class TransferForms implements Iterable<FormStatus> {
   private List<FormStatus> forms;
   private Map<String, FormStatus> formsIndex = new HashMap<>();
+  private Map<String, String> lastPullCursorsByFormId;
   private final List<Runnable> onChangeCallbacks = new ArrayList<>();
 
-  private TransferForms(List<FormStatus> forms) {
+  private TransferForms(List<FormStatus> forms, Map<String, String> lastPullCursorsByFormId) {
     this.forms = forms;
+    this.lastPullCursorsByFormId = lastPullCursorsByFormId;
     rebuildIndex();
   }
 
   /**
    * Factory of empty {@link TransferForms} instances
    */
   public static TransferForms empty() {
-    return new TransferForms(Collections.emptyList());
+    return new TransferForms(Collections.emptyList(), Collections.emptyMap());
   }
 
   /**
    * Factory of {@link TransferForms} instances that begin with the given
    * list of {@link FormStatus} instances
    */
   public static TransferForms from(List<FormStatus> forms) {
-    return new TransferForms(forms);
+    return new TransferForms(forms, Collections.emptyMap());
   }
 
   private static String getFormId(FormStatus form) {
     return form.getFormDefinition().getFormId();
   }
 
+  public static TransferForms of(FormStatus... forms) {
+    return new TransferForms(Arrays.asList(forms), Collections.emptyMap());
+  }
+
   /**
    * Replaces the current list of {@link FormStatus} instances with
-   * the incoming list.
+   * the incoming list, and reads any saved cursor from a previous pull
    */
-  public void load(List<FormStatus> forms) {
+  public void load(List<FormStatus> forms, BriefcasePreferences preferences) {
     this.forms = forms;
+    this.lastPullCursorsByFormId = forms.stream()
+        .map(form -> Pair.of(form.getFormId(), preferences.nullSafeGet(form.getFormId() + "-last-cursor")))
```

Comment:
"-last-cursor" appears in three places. Did you consider making it a constant?

Reply:
You're right :)
Thanks for pointing it out